src/main/java/com/example/onlinebookstore/repository/book/spec/PriceSpecificationProvider.java

```diff
@@ -0,0 +1,24 @@
+package com.example.onlinebookstore.repository.book.spec;
+
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String ATTRIBUTE = "price";
+
+    @Override
+    public String getKey() {
+        return ATTRIBUTE;
+    }
+
+    @Override
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder) -> root
+                .get(ATTRIBUTE)
+                .in(Arrays.stream(params).toArray());
```

Comment:
price IN? It is impossible to use, you should have some range (min-max values for price)